warehouse/query-core/src/main/java/datawave/query/iterator/logic/IndexIterator.java

```diff
@@ -95,6 +96,7 @@ public IndexIterator build() {
     public static final String INDEX_FILTERING_CLASSES = "indexfiltering.classes";
     
     protected SortedKeyValueIterator<Key,Value> source;
+    protected LimitedSortedKeyValueIterator limitedSource;
```

Comment:
Would it be possible to do this in a way that we only interact with `limitedSource` within this class, and remove `source` as a field -or- simply assign the `LimitedSortedKeyValueIterator` to the `source` field?

Reply:
we don't actually want to limit the source generally, just for the aggregator. This is to avoid re-seeking the source between aggregator calls

Reply:
got it. thanks.